purchase_delivery_split_date/tests/test_purchase_delivery.py

```diff
@@ -147,3 +150,37 @@ def test_purchase_line_date_change_split_picking(self):
             "If I change the other line to the same date as the first, "
             "both moves must be in the same picking",
         )
+
+    def test_purchase_line_created_afer_confirm(self):
+        """Check new line created when order is confirmed.
+
+        When a new line is added on an already `purchased` order
+        If it is planned for a non yet existing date in the purchase, a
+        new picking should be created.
+
+        """
+        self.po.button_confirm()
+        self.assertEqual(self.po.state, "purchase")
+        new_date = "2016-01-30"
+        moves_before = self.env["stock.move"].search(
+            [("purchase_line_id", "in", self.po.order_line.ids)]
+        )
+        self.assertEqual(len(moves_before.mapped("picking_id")), 1)
+        self.po.order_line = [
```

Comment:
cosmetic: an explicit call to create line or write order would be more clear IMO.